pkg/apis/core/resource_test.go

```diff
@@ -0,0 +1,178 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package core
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/davecgh/go-spew/spew"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+func TestResourceNameString(t *testing.T) {
+	testCases := []struct {
+		resourceName ResourceName
+		expected     string
+	}{
+		{
+			resourceName: ResourceName("secret"),
+			expected:     "secret",
+		},
+	}
+
+	for _, tc := range testCases {
+		if tc.resourceName.String() != tc.expected {
+			t.Errorf("expected ResourceName string to be '%v', but got '%v' instead", tc.expected, tc.resourceName.String())
```

Comment:
Instead of `'%v'` better write `%q`. It does escaping correctly if necessary.

Reply:
nit: as there is only one test, I would also not mind to drop the loop.

Reply:
There are some more `'%v'` below.

Reply:
Ack.
- replaced `'%v`s with `%q`
- dropped the loop